### PR TITLE
Record the prirority of selected preemption victims

### LINKED ISSUE #122046: Add PodPriority to preemption_victims metrics in scheduler
### What would you like to be added?

I propose enhancing the scheduler's preemption_victims metrics by including PodPriority information. This addition will provide more detailed insights into the preemption process, helping to understand which pods are being preempted and why, based on their priority.
https://github.com/kubernetes/kubernetes/blob/d61cbac69aae97db1839bd2e0e86d68f26b353a7/pkg/scheduler/metrics/metrics.go#L86-L94

### Why is this needed?

Including PodPriority in the preemption_victims metrics is essential for providing a more complete view of the scheduler's decision-making process. Currently, the metrics provide the count of preempted pods but lack detailed information about their priority. With this enhancement, administrators and developers can better analyze preemption patterns and priorities, leading to more informed decisions and potentially improving cluster efficiency and scheduling fairness.

### PR DESCRIPTION
**Pull request by utam0k**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Record the priority of selected preemption victims.

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubernetes/issues/122046

#### Does this PR introduce a user-facing change?

```release-note
`preemption_victims` metrics will also record the `spec.priorityClassName` of pods.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by utam0k** [comment]
/assign @sanposhiho 

Could you review this PR?

**Comment by dashpole** [comment]
/assign
for sig-instrumentation
/triage accepted

File: pkg/scheduler/metrics/metrics.go [hunk 1]
```diff
@@ -83,15 +83,15 @@ var (
...
 		&metrics.HistogramOpts{
 			Subsystem: SchedulerSubsystem,
 			Name:      "preemption_victims",
-			Help:      "Number of selected preemption victims",
+			Help:      "Number of selected preemption victims, by the priority of the preemptor pod.",
 			// we think #victims>64 is pretty rare, therefore [64, +Inf) is considered a single bucket.
 			Buckets:        metrics.ExponentialBuckets(1, 2, 7),
 			StabilityLevel: metrics.STABLE,
-		})
+		}, []string{"priority"})
```
Comment (kerthcet) [inline pkg/scheduler/metrics/metrics.go:94]: This is a STABLE metric, I think this is not acceptable, right? @logicalhan

Comment (utam0k) [reply pkg/scheduler/metrics/metrics.go:94]: Oh, sorry, I didn't know that. Is there relevant documentation? Can't I just add a tag?

Comment (dashpole) [reply pkg/scheduler/metrics/metrics.go:94]: I believe @kerthcet is correct.  From https://kubernetes.io/docs/reference/instrumentation/metrics/#list-of-stable-kubernetes-metrics:

> Stable metrics observe strict API contracts and no labels can be added or removed from stable metrics during their lifetime.

Comment (sanposhiho) [reply pkg/scheduler/metrics/metrics.go:94]: I completely didn't know that, good to know.

Comment (utam0k) [reply pkg/scheduler/metrics/metrics.go:94]: Thanks for your guide 🙏 Is adding the new metrics the only way to achieve https://github.com/kubernetes/kubernetes/issues/122046?

Comment (sanposhiho) [reply pkg/scheduler/metrics/metrics.go:94]: Yes, and I'm honestly not sure it deserves creating a new metric.
@kerthcet What do you think.

Comment (kerthcet) [reply pkg/scheduler/metrics/metrics.go:94]: Yes, keeping two similar metrics is not a good idea. Maybe we can back to this once we got similar feedbacks from the community.

Comment (utam0k) [reply pkg/scheduler/metrics/metrics.go:94]: As a user in production operations, it is a label I want.

Comment (sanposhiho) [reply pkg/scheduler/metrics/metrics.go:94]: +1 for @kerthcet.
I agree that it's valuable though, let's once close this PR while keeping https://github.com/kubernetes/kubernetes/issues/122046.
If we've got the same feedback, or we've got another feature request for this metric, then we can consider creating a new one to eventually replace this metric.

**Pull request closed without merging**